Refactor: Calendar

Plebysnacc opened the pull request:
This seems to be some work. I've added the component so y'all can take a look at it, but the actual functionality and API has to be wired in. Also have to change the code up to fix our needs.

Will probably take at least till the weekend.

- [x] debug the damn times again
- [x] create better tutorial views (moved to another issue)
- [x] make color dependent of topic or type
- [x] fix sticky headers colliding 
- [x] remove debug statements
- [x] remove drag and drop

Plebysnacc commented:
Got some of it working, will start with debugging time and creating better tutorial views tomorrow

dheidemann commented:
this looks much better than the previous version! Can I assists you somewhere?

Plebysnacc commented:
Hey, thanks. I'm currently recovering from surgery, so progress is slow. 

You can work on this if it's time critical. I think it's clear what I wanted to achieve with the new event dialog-tutorial ui. The logic did not change

If you want, we can hop into a call, and I'll give u a quick run down on what's needed here

dheidemann commented:
there is currently an issue when registering for an event it works, but in the table you need a reload before it becomes visible... dunno whats going on

Plebysnacc commented:
Want me to have a look at it?

Plebysnacc commented:
Oke started working on it, the bug is funny af lol

Plebysnacc commented:
Hello there! Tutorial bug is fixed. Reason for bug(s) was:

The tutorial query did not ask for event ID, so on registration changes the "new" registration had `eventID = 0` and thus did not make the table update its buttons 

Things I also changed:
- some variable renaming for better readability
- changed the loading handling, so the utilization bar does not flicker
- simplified some boolean evaluations

Plebysnacc commented:
Okay dokay, have a look at it mate

dheidemann reviewed (approved):
lets go ma guy
